### PR TITLE
Improve UDP-Device Error handling

### PR DESCRIPTION
**Pull request by Lord-Grey**
**Summary**

Improve on the error-handling and output for the UDP-Provider class.

In support of hyperion forum topic:
https://forum.hyperion-project.org/threads/new-user-trying-to-make-it-work.10643/#post-26971

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of web configuration, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing setups:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's body (e.g. `Fixes: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated (docs/docs/en)
- [ ] Related tests have been updated

**PLEASE DON'T FORGET TO ADD YOUR CHANGES TO CHANGELOG.MD**
- [ ] Yes, CHANGELOG.md is also updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Comment by Paulchen-Panther** [comment]
Teste ich morgen mit WLED. 👍

**Comment by Paulchen-Panther** [comment]
Works fine with ip and hostname.
Proof & Log:
led_hardware
wled

```
paulchen@paulchen-VirtualBox:~/Schreibtisch/hyperion/bin$ ./hyperiond -d
Qt: Session management error: Authentication Rejected, reason : None of the authentication protocols specified are supported and host-based authentication failed
2020-08-30T15:03:16.928 hyperiond MAIN         : <INFO> Set user data path to '/home/paulchen/.hyperion'
2020-08-30T15:03:17.075 hyperiond DAEMON       : <DEBUG> PythonInit.cpp:44:PythonInit() | Initializing Python interpreter
2020-08-30T15:03:17.292 hyperiond SETTINGSMGR  : <DEBUG> SettingsManager.cpp:107:SettingsManager() | Settings database initialized
2020-08-30T15:03:17.293 hyperiond DAEMON       : <INFO> CEC handler created
2020-08-30T15:03:17.303 hyperiond EFFECTFILES  : <INFO> 39 effects loaded from directory :/effects/
2020-08-30T15:03:17.307 hyperiond EFFECTFILES  : <INFO> 22 effect schemas loaded from directory :/effects/schema/
2020-08-30T15:03:17.308 hyperiond EFFECTFILES  : <INFO> New Effect path "/home/paulchen/.hyperion/custom-effects" created successfull
2020-08-30T15:03:17.332 hyperiond SETTINGSMGR  : <DEBUG> SettingsManager.cpp:107:SettingsManager() | Settings database initialized
2020-08-30T15:03:17.334 hyperiond BLACKBORDER  : <DEBUG> BlackBorderProcessor.cpp:64:handleSettingsUpdate() | Set mode to: default
2020-08-30T15:03:17.334 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | Blackborder detector: enabled
2020-08-30T15:03:17.349 hyperiond DAEMON       : <INFO> set screen capture device to 'x11'
2020-08-30T15:03:17.349 hyperiond X11GRABBER   : <DEBUG> Grabber.cpp:30:setVideoMode() | Set videomode to 0
2020-08-30T15:03:17.349 hyperiond DAEMON       : <INFO> X11 grabber created
2020-08-30T15:03:17.350 hyperiond V4L2:AUTO    : <DEBUG> Grabber.cpp:30:setVideoMode() | Set videomode to 0
2020-08-30T15:03:17.350 hyperiond V4L2:AUTO    : <INFO> Signal threshold set to: {12, 12, 12}
2020-08-30T15:03:17.350 hyperiond V4L2:AUTO    : <INFO> CEC detection is now disabled
2020-08-30T15:03:17.351 hyperiond V4L2:AUTO    : <INFO> Signal detection is now disabled
2020-08-30T15:03:17.351 hyperiond V4L2:AUTO    : <INFO> Signal detection area set to: 0.250000,0.250000 x 0.750000,0.750000
2020-08-30T15:03:17.351 hyperiond DAEMON       : <DEBUG> hyperiond.cpp:567:handleSettingsUpdate() | V4L2 grabber created
2020-08-30T15:03:17.351 hyperiond JSONSERVER   : <DEBUG> JsonServer.cpp:27:JsonServer() | Created instance
2020-08-30T15:03:17.352 hyperiond JSONSERVER   : <INFO> Started on port 19444
2020-08-30T15:03:17.353 hyperiond FLATBUFSERVE : <INFO> Started on port 19400
2020-08-30T15:03:17.354 hyperiond PROTOSERVER  : <INFO> Started on port 19445
2020-08-30T15:03:17.354 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:32:initServer() | Initialize Webserver
2020-08-30T15:03:17.355 hyperiond MAIN         : <INFO> start systray
2020-08-30T15:03:17.358 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:32:initServer() | Initialize Webserver
2020-08-30T15:03:17.364 hyperiond LEDDEVICE    : <INFO> Start LedDevice 'file'.
2020-08-30T15:03:17.365 hyperiond LEDDEVICE    : <DEBUG> LedDevice.cpp:133:init() | deviceConfig: [{"colorOrder":"rgb","currentLedCount":26,"hardwareLedCount":1,"latchTime":0,"output":"/dev/null","rewriteTime":0,"type":"file"}]
2020-08-30T15:03:17.366 hyperiond LEDDEVICE    : <DEBUG> LedDeviceFile.cpp:51:open() | QIODevice::WriteOnly, /dev/null
2020-08-30T15:03:17.378 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | Smoothing: enabled
2020-08-30T15:03:17.380 hyperiond FLATBUFCONN  : <INFO> Connecting to Hyperion: 127.0.0.1:19401
2020-08-30T15:03:17.380 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:90:handleSettingsUpdate() | Apply Webserver settings
2020-08-30T15:03:17.380 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:108:handleSettingsUpdate() | Set document root to: :/webconfig
2020-08-30T15:03:17.381 hyperiond WEBSERVER    : <INFO> Started on port 8090 name 'Hyperion Webserver'
2020-08-30T15:03:17.386 hyperiond EFFECTENGINE : <INFO> Run effect "Rainbow swirl fast" on channel 0
2020-08-30T15:03:17.387 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Rainbow swirl fast], smoothCfg [2]
2020-08-30T15:03:17.387 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/EFFECT' with priority 0 as inactive
2020-08-30T15:03:17.387 hyperiond HYPERION     : <INFO> Initial foreground effect 'Rainbow swirl fast' started
2020-08-30T15:03:17.400 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:90:handleSettingsUpdate() | Apply Webserver settings
2020-08-30T15:03:17.401 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:108:handleSettingsUpdate() | Set document root to: :/webconfig
2020-08-30T15:03:17.419 hyperiond EFFECTENGINE : <INFO> Run effect "Warm mood blobs" on channel 254
2020-08-30T15:03:17.420 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Warm mood blobs], smoothCfg [2]
2020-08-30T15:03:17.420 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/EFFECT' with priority 254 as inactive
2020-08-30T15:03:17.421 hyperiond HYPERION     : <INFO> Inital background effect 'Warm mood blobs' started
2020-08-30T15:03:17.422 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/GRABBER' with priority 250 as inactive
2020-08-30T15:03:17.423 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | Framegrabber: enabled
2020-08-30T15:03:17.424 hyperiond BOBLIGHT     : <DEBUG> BoblightServer.cpp:28:BoblightServer() | Instance created
2020-08-30T15:03:17.424 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | LED device: enabled
2020-08-30T15:03:17.424 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:174:handleSettingsUpdate() | Setup SSL certificate
2020-08-30T15:03:17.425 hyperiond WEBSERVER    : <DEBUG> WebServer.cpp:190:handleSettingsUpdate() | Setup private SSL key
2020-08-30T15:03:17.426 hyperiond WEBSERVER    : <INFO> Started on port 8092 name 'Hyperion Webserver'
2020-08-30T15:03:17.519 hyperiond X11          : <DEBUG> GrabberWrapper.cpp:48:start() | Grabber start()
2020-08-30T15:03:17.521 hyperiond HYPERION     : <INFO> Hyperion instance 'First LED Hardware instance' has been started
2020-08-30T15:03:17.653 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:238:setInputImage() | Priority 0 is now active
2020-08-30T15:03:17.653 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:333:setCurrentTime() | Set visible priority to 0
2020-08-30T15:03:17.657 hyperiond X11GRABBER   : <INFO> Update of screen resolution: [0x0]  to [1416x770]
2020-08-30T15:03:17.657 hyperiond X11GRABBER   : <INFO> Using XRender for grabbing
2020-08-30T15:03:17.657 hyperiond X11GRABBER   : <INFO> Update output image resolution: [0x0]  to [177x96]
2020-08-30T15:03:17.659 hyperiond X11GRABBER   : <INFO> Capture interface is now enabled
2020-08-30T15:03:17.674 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:238:setInputImage() | Priority 250 is now active
2020-08-30T15:03:17.695 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:198:setInput() | Priority 254 is now active
2020-08-30T15:03:20.563 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:298:setCurrentTime() | Timeout clear for priority 0
2020-08-30T15:03:20.814 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:333:setCurrentTime() | Set visible priority to 250
2020-08-30T15:03:45.430 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:192.168.0.90
2020-08-30T15:03:45.431 hyperiond WEBSOCKET    : <DEBUG> JsonAPI.cpp:107:handleInstanceSwitch() | Client '::ffff:192.168.0.90' switch to Hyperion instance 0
2020-08-30T15:04:26.327 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:204:sendClose() | send close: 1000 
2020-08-30T15:04:27.883 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:192.168.0.90
2020-08-30T15:04:27.883 hyperiond WEBSOCKET    : <DEBUG> JsonAPI.cpp:107:handleInstanceSwitch() | Client '::ffff:192.168.0.90' switch to Hyperion instance 0
2020-08-30T15:08:04.423 hyperiond LEDDEVICE    : <DEBUG> LedDeviceFile.cpp:77:close() | File: /dev/null
[hyperiond LedDeviceWrapper] <INFO> LedDevice 'file' closed
2020-08-30T15:08:04.424 hyperiond LEDDEVICE    : <INFO> Start LedDevice 'wled'.
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:56:init() | 
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDevice.cpp:133:init() | deviceConfig: [{"colorOrder":"rgb","currentLedCount":30,"hardwareLedCount":30,"host":"WLED-RGBW.local","latchTime":0,"type":"wled"}]
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:64:init() | DeviceType   : wled
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:65:init() | LedCount     : 30
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:66:init() | ColorOrder   : rgb
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:67:init() | LatchTime    : 0
2020-08-30T15:08:04.425 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:109:initRestAPI() | 
2020-08-30T15:08:04.426 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:120:initRestAPI() | [1]
2020-08-30T15:08:04.427 hyperiond LEDDEVICE    : <DEBUG> LedDevice.cpp:133:init() | deviceConfig: [{"colorOrder":"rgb","currentLedCount":30,"hardwareLedCount":30,"host":"WLED-RGBW.local","latchTime":0,"port":19446,"type":"wled"}]
2020-08-30T15:08:04.430 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | LED device: disabled
2020-08-30T15:08:04.435 hyperiond LEDDEVICE    : <DEBUG> ProviderUdp.cpp:52:init() | Successfully resolved IP-address (192.168.0.60) for hostname (WLED-RGBW.local).
2020-08-30T15:08:04.435 hyperiond LEDDEVICE    : <DEBUG> ProviderUdp.cpp:74:init() | UDP socket will write to 192.168.0.60:19446
2020-08-30T15:08:04.435 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:98:init() | Hostname/IP  : WLED-RGBW.local
2020-08-30T15:08:04.435 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:99:init() | Port         : 19446
2020-08-30T15:08:04.435 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:103:init() | [1]
2020-08-30T15:08:04.436 hyperiond LEDDEVICE    : <DEBUG> LedDeviceWled.cpp:132:powerOn() | 
2020-08-30T15:08:04.436 hyperiond LEDDEVICE    : <DEBUG> ProviderRestApi.cpp:153:put() | PUT: [http://wled-rgbw.local/json/state] [{"on":true}]
2020-08-30T15:08:04.508 hyperiond LEDDEVICE    : <DEBUG> ProviderRestApi.cpp:182:getResponse() | Reply.httpStatusCode [200]
2020-08-30T15:08:04.508 hyperiond COMPONENTREG : <DEBUG> ComponentRegister.cpp:36:setNewComponentState() | LED device: enabled
2020-08-30T15:08:37.233 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:75:setSourceAutoSelectEnabled() | Source auto select is now disabled
2020-08-30T15:08:37.234 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:333:setCurrentTime() | Set visible priority to 254
2020-08-30T15:09:01.290 hyperiond EFFECTENGINE : <INFO> Run effect "Rainbow swirl fast" on channel 0
2020-08-30T15:09:01.290 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Rainbow swirl fast], smoothCfg [0]
2020-08-30T15:09:01.290 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:158:registerInput() | Register new input 'System/EFFECT' with priority 0 as inactive
2020-08-30T15:09:01.291 hyperiond EFFECTENGINE : <INFO> Run effect "Warm mood blobs" on channel 254
2020-08-30T15:09:01.291 hyperiond EFFECTENGINE : <DEBUG> EffectEngine.cpp:187:runEffectScript() | Start the effect: name [Warm mood blobs], smoothCfg [0]
2020-08-30T15:09:01.355 hyperiond EFFECTENGINE : <INFO> effect finished
2020-08-30T15:09:01.587 hyperiond EFFECTENGINE : <INFO> Previous line repeats 1 times
2020-08-30T15:09:01.587 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:238:setInputImage() | Priority 0 is now active
2020-08-30T15:09:04.563 hyperiond HYPERION     : <DEBUG> PriorityMuxer.cpp:298:setCurrentTime() | Timeout clear for priority 0
2020-08-30T15:09:18.851 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:204:sendClose() | send close: 1000 
2020-08-30T15:09:20.359 hyperiond WEBSOCKET    : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:192.168.0.90
2020-08-30T15:09:20.359 hyperiond WEBSOCKET    : <DEBUG> JsonAPI.cpp:107:handleInstanceSwitch() | Client '::ffff:192.168.0.90' switch to Hyperion instance 0
```

**Review by Paulchen-Panther — APPROVED** [review]

File: libsrc/leddevice/dev_net/ProviderUdp.cpp [hunk 1]
```diff
@@ -45,41 +45,42 @@ bool ProviderUdp::init(const QJsonObject &deviceConfig)
...
-			QHostInfo info = QHostInfo::fromName(host);
-			if (info.addresses().isEmpty())
+			QHostInfo hostInfo = QHostInfo::fromName(host);
+			if ( hostInfo.error() == QHostInfo::NoError )
 			{
-				Debug( _log, "Failed to parse [%s] as a hostname.", deviceConfig["host"].toString().toStdString().c_str());
-				QString errortext = QString ("Invalid target address [%1]!").arg(host);
-				this->setInError ( errortext );
-				return false;
+				_address = hostInfo.addresses().first();
```
Comment (m-seker) [inline libsrc/leddevice/dev_net/ProviderUdp.cpp:55]: Is ````addresses()```` guaranteed to be non empty ?

**Review by m-seker — APPROVED** [review]

File: libsrc/leddevice/dev_net/ProviderUdp.cpp (hunk 1 above)
Comment (Lord-Grey) [reply libsrc/leddevice/dev_net/ProviderUdp.cpp:55]: Yes. The code is only executed, if no error during Hostname resolution occurred; I.e. it is existing.

QHostInfo hostInfo = QHostInfo::fromName(host);
 			if ( hostInfo.error() == QHostInfo::NoError )

**Pull request merged**